### PR TITLE
DOC: Improve truncate docstring (#17763)

### LINKED ISSUE #17763: DateFrame.truncate is not equivalent to slicing as mentioned in docs
#### Code Sample, a copy-pastable example if possible

```python
>>> index = pd.date_range('2016-1-1', '2016-2-1', freq='s')
>>> df = pd.DataFrame(index=d, data={'a':1})  
>>> df.loc['2016-1-5 05':'2016-1-10']                  
                     a
2016-01-05 00:00:00  1
2016-01-05 00:00:01  1
2016-01-05 00:00:02  1
...
2016-01-10 23:59:57  1
2016-01-10 23:59:58  1
2016-01-10 23:59:59  1

>>> df.truncate('2016-1-5', '2016-1-10')
                     a
2016-01-05 00:00:00  1
2016-01-05 00:00:01  1
2016-01-05 00:00:02  1
...
2016-01-09 23:59:58  1
2016-01-09 23:59:59  1
2016-01-10 00:00:00  1
```
#### Problem description
The [documentation](http://pandas.pydata.org/pandas-docs/stable/timeseries.html#truncating-fancy-indexing) for `truncate` has misleading information. `truncate` and slicing are not the same. Slicing returns all partially matching dates while truncate stops at an exact datetime - it assumes 0's for any unspecified date component.

Also, truncate can take datetime and partial string datetimes which, to me, isn't apparent in the docstrings.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.1.final.0
python-bits: 64
OS: Darwin
OS-release: 15.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.20.3
pytest: 3.0.7
pip: 9.0.1
setuptools: 35.0.2
Cython: 0.25.2
numpy: 1.13.1
scipy: 0.19.0
xarray: None
IPython: 6.0.0
sphinx: 1.5.5
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: 1.2.0
tables: 3.4.2
numexpr: 2.6.2
feather: None
matplotlib: 2.0.2
openpyxl: 2.4.7
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.7.3
bs4: 4.6.0
html5lib: 0.9999999
sqlalchemy: 1.1.9
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: 0.3.0.post

</details>

### PR DESCRIPTION
**Pull request by reidy-p**
- [ ] closes #17763

### CODE REVIEW

File: doc/source/timeseries.rst [hunk 1]
```diff
@@ -657,11 +657,14 @@ With no defaults.
...
-A ``truncate`` convenience function is provided that is equivalent to slicing:
+A ``truncate`` convenience function is provided that is similar to slicing.
+Note that ``truncate`` assumes a 0 value for any unspecified date component
+in a ``DatetimeIndex`` in contrast to slicing which returns any partially
+matching dates:
 
 .. ipython:: python
 
-   ts.truncate(before='10/31/2011', after='12/31/2011')
+   ts.truncate(before='2011', after='2012')
```
Comment (jorisvandenbossche) [inline doc/source/timeseries.rst]: maybe add the equivalent slicing example as well to show it is not the same ?

File: pandas/core/generic.py [hunk 2]
```diff
@@ -6330,17 +6330,64 @@ def truncate(self, before=None, after=None, axis=None, copy=True):
...
-            Truncate after index value
-        axis : the truncation axis, defaults to the stat axis
+        before : date, string, int
+            Truncate all rows before this index value
+        after : date, string, int
+            Truncate all rows after this index value
+        axis : {0 or 'index', 1 or 'columns'}
+            * 0 or 'index': apply truncation to rows
+            * 1 or 'columns': apply truncation to columns
+            Default is stat axis for given data type (0 for Series and
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: blank line above this one

File: pandas/core/generic.py [hunk 3]
```diff
@@ -6330,17 +6330,64 @@ def truncate(self, before=None, after=None, axis=None, copy=True):
...
+        b  2  7  12
+        c  3  8  13
+        d  4  9  14
+
+        The index values in ``truncate`` can be datetimes or string
+        dates. Note that ``truncate`` assumes a 0 value for any unspecified
+        date component in a ``DatetimeIndex`` in contrast to slicing which
+        returns any partially matching dates.
+
+        >>> dates = pd.date_range('2016-1-1', '2016-2-1', freq='s')
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: nitpick: I prefer to use iso dates in our docs (apart for examples specifically on date string parsing), so "2016-01-01" etc

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks! Added a few comments.

In the docstring of `truncate`, can you also edit the first line of the docstring to replace 'NDFrame' with "DataFrame/Series" ?

**Comment by jreback** [comment]
thanks @reidy-p

**Pull request merged**